pkg/agent/util/dump_linux.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+	"k8s.io/utils/exec"
+
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsctl"
+	generalutil "github.com/vmware-tanzu/antrea/pkg/util"
+)
+
+type dumper struct {
+	factory  ofctlClientFactory
+	fs       afero.Fs
+	executor exec.Interface
+}
+
+func (d *dumper) DumpLog(basedir string) error {
+	generalDumper := generalutil.NewDumper(d.fs, d.executor)
+	if err := generalDumper.DumpLog(path.Join(basedir, "logs", "agent"), "/var/log/antrea", "antrea-agent"); err != nil {
+		return err
+	}
+	return generalDumper.DumpLog(path.Join(basedir, "logs", "ovs"), "/var/log/antrea", "ovs")
+}
+
+func (d *dumper) DumpFlows(basedir string) error {
+	brs, err := ovsctl.ListBridges(d.executor)
+	if err != nil {
+		return fmt.Errorf("error when collecting ovs bridge info: %w", err)
+	}
+	err = d.fs.Mkdir(filepath.Join(basedir, "flows"), os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("error when creating flows output dir: %w", err)
+	}
+	for _, br := range brs {
+		flows, err := d.factory(br).DumpFlows()
+		if err != nil {
+			return fmt.Errorf("error when dumping flows on bridge %s: %w", br, err)
+		}
+		err = afero.WriteFile(d.fs, filepath.Join(basedir, "flows", br), []byte(strings.Join(flows, "\n")), 0644)
+		if err != nil {
+			return fmt.Errorf("error when creating flows output file: %w", err)
+		}
+	}
+	return nil
+}
+
+func (d *dumper) DumpHostNetworkInfo(basedir string) error {
+	if err := d.dumpIPTables(basedir); err != nil {
+		return err
+	}
+	if err := d.dumpIPToolInfo(basedir); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *dumper) dumpIPTables(basedir string) error {
+	iptablesOutput, err := d.executor.Command("iptables-save").CombinedOutput()
```

Comment:
suggest to reuse pkg/util/iptables by adding Save method.

Reply:
Added.